serenity/serenity.hpp

```diff
@@ -26,6 +26,41 @@ class Consumer : public BusSocket {
 };
 
 
+/**
+ * This consumer requires that all producents will run consume on
+ * this instance. (Even during error flow).
+ */
+template<typename T>
+class MultiConsumer : public Consumer<T> {
```

Comment:
I would name it more like "SyncMultiConsume".
We can do multiconsumption without this class - but here we are waiting for all products.